internal/controlplane/handlers_oauth.go

```diff
@@ -55,13 +58,20 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 	entityCtx := engine.EntityFromContext(ctx)
 	projectID := entityCtx.Project.ID
 
+	var provider string
+	if req.GetContext().GetProvider() == "" {
+		provider = defaultProvider
+	} else {
+		provider = req.GetContext().GetProvider()
+	}
```

Comment:
If we're removing the default provider, wouldn't it make sense that we first do a `CreateProvider` call and THEN do the `GetAuthorizationURL`? The logic then wouldn't change here.

Reply:
My thinking is that if a provider requires credentials, then we shouldn't save the provider in the DB until we have credential for it. Otherwise we'll end up with providers that aren't usable.

Reply:
I like this idea. Let's go for that. Thanks!